lib/Targets/AIEVecToCpp/TranslateAIEVecToCpp.cpp

```diff
@@ -357,15 +359,15 @@ static LogicalResult createLinearizedAccess(CppEmitter &emitter, Value source,
   std::string paramPart; // for dynamic shaped dims
 
   SmallVector<std::string, 4> accessVec;
-  for (int dim = memRefType.getRank() - 1; dim >= 0; --dim) {
+  for (int64_t dim = memRefType.getRank() - 1; dim >= 0; --dim) {
     // All the indices in the access expression must already be emitted
     if (!emitter.hasValueInScope(indices[dim]))
       return failure();
 
     // Form the access string for this dimension
     std::string cur;
     if (!paramPart.empty())
-      cur = paramPart + "*";
+      cur += paramPart + "*";
```

Comment:
String concatenation results in allocation of unnecessary temporary strings; consider using 'operator+=' or 'string::append()' instead